### PR TITLE
Add spread information for each pair

### LINKED ISSUE #86: Market: Add "spread" information for each pair
The spread can be calculated by taking the best price for the opposite pair (direction) and calculate as

spread = bestPrice - (1 / oppositeBestPrice)
spread_percentage = spread / bestPrice

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #86

Still, there seems to be some issue with the Math formula, since some spreads are negative

image

### CODE REVIEW

File: apis/cosmos-reducers.js [hunk 1]
```diff
@@ -675,6 +675,7 @@ export function activeInstrumentReducer(instrument) {
   return {
     date: instrument.last_traded,
     denom: denomLookup(instrument.destination),
+    counterDenom: denomLookup(instrument.source),
```
Comment (mdyring) [inline apis/cosmos-reducers.js:678]: Suggest we rename denom and counterDenom:

    source: denomLookup(instrument.source),
    destination: denomLookup(instrument.destination),

File: components/swap/TableActiveInstruments.vue [hunk 2]
```diff
@@ -56,15 +57,41 @@ export default {
         this.selectCurrency(currenciesTabsArray[0].value)
       return currenciesTabsArray
     },
+    activeInstrumentsWithSpread() {
+      return (
+        this.activeInstruments
+          .map((instrument) => {
+            const oppositeBestPrice = this.activeInstruments.find(
+              ({ denom }) => denom === instrument.counterDenom
```
Comment (mdyring) [inline components/swap/TableActiveInstruments.vue:65]: Finding the opposite instrument should check that
1) current instrument source === opposite instrument destination
2) current instrument destination === opposite instrument source

(assuming the rename suggested above was done)

Comment (Bitcoinera) [reply components/swap/TableActiveInstruments.vue:65]: Well, this is what is happening here. I did the Math with the formula and I still get negative numbers. Look at this one example:

```
"source": "edkk",
"destination": "echf",
"best_price": "0.142517421779270760"

"source": "echf",
"destination": "edkk",
"best_price": "6.942880643556953522"
```

spread = bestPrice - (1 / oppositeBestPrice) = 6.94 - (1 / 0.14) = 6.94 - 7.14 = -0.2

File: apis/cosmos-reducers.js (hunk 1 above)
Comment (Bitcoinera) [reply apis/cosmos-reducers.js:678]: somehow I find this naming more intuitive. I can tell "denom" is the base value here, more intuitively than "destination". But I can change it if this is very important to you

**Comment by mdyring** [comment]
Working now, just a couple of minor fixes:
1) The wrong "oppositeInstrument" was used
2) Fixed formula - the one provided in the original issue was wrong

Seems like sorting by spread does not work however.. Maybe it can just be disabled, since it does not seem like a useful thing to sort on?

image

**Comment by Bitcoinera** [comment]
Amazing

**Pull request merged**